bridge_adaptivity/module/models.py

```diff
@@ -13,12 +15,33 @@
 from ordered_model.models import OrderedModel
 
 from bridge_lti.models import BridgeUser, LtiConsumer, LtiUser, OutcomeService
-from module import ENGINE
-
+from common import utils
 
 log = logging.getLogger(__name__)
 
 
+def _discover_engines():
+    engines = []
+    for name in os.listdir(os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), 'engines'
+    )):
+        if name.startswith('engine_') and name.endswith('.py'):
+            engines.append((name, name[len('engine_'):-len('.py')]))
+    return engines
+
+
+def _get_engine_driver(engine):
+    driver = None
+    engine_module = importlib.import_module('module.engines.{}'.format(engine))
+    for attr in inspect.getmembers(engine_module):
+        if attr[0].startswith('Engine'):
+            driver = attr[1]
+    return driver
+
+
+ENGINES = _discover_engines()
```

Comment:
as far as I remember we discussed that user should be able to add new engines without server reloading? isn't it? 

If I correctly understand what's going on here - user will not be able to add new one engine on the fly. 

correct me If I missed something.

Reply:
New Engine could be created dynamically `_discover_engines()` returns engines libs with the driver API implementation.